reference-architecture/azure-ansible/3.6/allinone.sh

```diff
@@ -431,6 +431,213 @@ cat <<EOF > /home/${AUSERNAME}/postinstall.yml
 
 EOF
 
+cat > /home/${AUSERNAME}/ssovars.yml <<EOF
+---
+  sso_username: ${AUSERNAME}
+  sso_project: "sso"
+  sso_password: ${PASSWORD}
+  sso_domain:   ${WILDCARDNIP}
+  hostname_https: "login.{{sso_domain}}"
+  api_master:   ${WILDCARDFQDN}
+EOF
+
+cat > /home/${AUSERNAME}/setup-sso.yml <<EOF
+---
+- hosts: masters[0]
+  vars_files:
+    - ssovars.yml
+  vars:
+    description: "SSO Setup"
+    create_data:
+        clientId: "openshift"
+        name:     "OpenShift"
+        description: "OpenShift Console Authentication"
+        enabled: true
+        protocol: "openid-connect"
+        clientAuthenticatorType: "client-secret"
+        directAccessGrantsEnabled: true
+        redirectUris: ["https://{{api_master}}:8443/*"]
+        webOrigins: []
+        publicClient: false
+        consentRequired: false
+        frontchannelLogout: false
+        standardFlowEnabled: true
+  tasks:
+  - debug:
+      msg: "Domain: {{sso_domain}}"
+  - set_fact: idm_dir="/home/{{sso_username}}/{{sso_project}}"
+  - debug:
+      msg: "Idm dir {{ idm_dir }}"
+  - name: Install Java
+    yum:
+      name: java-1.8.0-openjdk
+      state: latest
+  - name: Cleanup old idm directory
+    file:
+      state: absent
+      path: "{{idm_dir}}"
+  - name: Create new idm directory
+    file: 
+      state: directory
+      path: "{{idm_dir}}"
+  - name: Delete service account
+    command: oc delete service account "{{sso_project}}-service_account"
+    ignore_errors: yes
+    register: result
+    failed_when: 
+      - "result.rc > 10"
+  - name: Delete Secret
+    command: oc delete secret "{{sso_project}}-app-secret"
+    ignore_errors: yes
+    register: result
+    failed_when: 
+      - "result.rc > 10"
+  - name: Delete Old Project
+    command: oc delete project "{{sso_project}}"
+    ignore_errors: yes
+    register: result
+    failed_when: 
+      - "result.rc > 10"
+  - name: Pause for cleanup of old install
+    pause:
+      minutes: 2
+  - set_fact: sso_projectid="{{sso_project}}"
+  - set_fact: idm_xpassword="Xp-{{sso_password}}"
+  - name: Create Openshift Project for SSO
+    command: oc new-project "{{sso_project}}"
+  - name: Create Service Account
+    command: "oc create serviceaccount {{sso_project}}-service-account"
```

Comment:
Just in case, I'll add the "-n {{ sso_project }}" to ensure the service account is created in the proper project